pkg/actuators/machine/reconciler.go

```diff
@@ -187,11 +188,48 @@ func (r *Reconciler) update() error {
 
 	klog.Infof("Updated machine %s", r.machine.Name)
 
-	r.machineScope.setProviderStatus(newestInstance, conditionSuccess())
+	domainNames, err := r.getCustomDomainFromDHCP(newestInstance.VpcId)
```

Comment:
I don't thins this should be a parameter for `setProviderStatus`. I think this should be called from within `extractNodeAddresses` or `setProviderStatus`. i.e they should never run with out this.